core/flyout_horizontal.ts

```diff
@@ -236,31 +244,31 @@ export class HorizontalFlyout extends Flyout {
 
     for (let i = 0, item; item = contents[i]; i++) {
       if (item.type === 'block') {
-        const block = item.block;
-        const allBlocks = block!.getDescendants(false);
+        const block = item.block as BlockSvg;
```

Comment:
Alternatively/safer:

if (!item.block) throw new Error(flyout item with type block must contain property block)

Reply:
I think this is actually validated in an earlier parsing step. There may be a better typing solution here so that we know that if type is `''block'`, it has the property `block`.

Reply:
yeah, I think you can use the type guard method strategy:

```
function isBlock(item: FlyoutItem): item is BlockSvg {
  return item.type === 'block';
}
```


then check `isBlock(item)` instead of `item.type === block`

but after writing all that out, i wonder why bother checking the type at all, and just check `if item.block` directly?

Reply:
You mean with an instanceof check? I guess we could, as long as we assume that there will never be other cases where you have an item that has a block in it. Which is probably safe, or at least something we could deal with if/when it becomes an issue.

Reply:
Fair point though, that if the goal is to actually have type safety, casually casting doesn't do it.

Reply:
I don't think we need an instanceof check. Initially I thought we could use a type guard but I don't think that works because we need to assert `item.block` is a `BlockSvg`, not `item`.

I'm saying instead of checking `if (item.type === 'block')`, just check `if (item.block)`. If `item.block` is not null, we already know it must be a `BlockSvg` (without using an instanceof check, just because of the type of `FlyoutItem`). So I don't think we need the `item.type` at all here. The only two options for the type are button and block so I don't think we need to worry about other types also having a block property.